### PR TITLE
Display cluster name for each connection & add autoconnect 

### PR DESCRIPTION
**Pull request by gzdunek**
Also fixes a bug when connecting to server in a leaf cluster

### CODE REVIEW

File: packages/teleterm/src/ui/DocumentGateway/useDocumentGateway.ts [hunk 1]
```diff
@@ -75,6 +73,12 @@ export default function useGateway(doc: types.DocumentGateway) {
     }
   }, [disconnectAttempt.status]);
 
+  useEffect(() => {
+    if (cluster.connected) {
+      createGateway();
+    }
+  }, [cluster.connected]);
+
```
Comment (ravicious) [inline packages/teleterm/src/ui/DocumentGateway/useDocumentGateway.ts:81]: What kind of scenario is this + the change in the `reconnect` above are going to handle?

File: packages/teleterm/src/ui/ServerConnect/useServerConnect.ts [hunk 2]
```diff
@@ -23,8 +23,10 @@ export default function useServerConnect({ serverUri, onClose }: Props) {
   const logins = cluster?.loggedInUser?.sshLoginsList || [];
 
   const connect = (login: string) => {
+    const rootCluster =
+      ctx.clustersService.findRootClusterByResource(serverUri);
     const documentsService = ctx.workspacesService.getWorkspaceDocumentService(
-      cluster.uri
+      rootCluster.uri
```
Comment (ravicious) [inline packages/teleterm/src/ui/ServerConnect/useServerConnect.ts:29]: Yeah, I'm still kinda conflicted whether we should convert a resource URI to a root cluster URI this way. The root cluster URI seems to be baked into every other URI, so theoretically we should be able to get it just from operations on that string alone, without having to use `clustersService`. My main concern is that in other places it'll require pulling in that `clustersService` which can unnecessarily complicate tests.

We might think about something like `routing.extractRootClusterUri` in the future. For now I added just `ensureRootClusterUri` there which takes any kind of cluster URI and converts it to a root cluster URI.

**Review by ravicious — APPROVED** [review]

File: packages/teleterm/src/ui/ServerConnect/useServerConnect.ts (hunk 2 above)
Comment (gzdunek) [reply packages/teleterm/src/ui/ServerConnect/useServerConnect.ts:29]: Good point, I will add it to our master ticket

File: packages/teleterm/src/ui/DocumentGateway/useDocumentGateway.ts (hunk 1 above)
Comment (gzdunek) [reply packages/teleterm/src/ui/DocumentGateway/useDocumentGateway.ts:81]: When you click on a connection it opens new tab with text: Click to reconnect, even when you are connected to the cluster (and in `useReconnect` is just changes document status).
After this change, clicking on a connection will try to make the _connection_ automatically and open a tab.

**Pull request merged**